src/functions/types.ts

```diff
@@ -93,15 +93,20 @@ type UnknownRuntimeType = any;
 
 type TypedObjectFunctionInputRuntimeType<
   Param extends TypedObjectParameterDefinition,
-> =
-  & {
+> = Param["additionalProperties"] extends false ? {
     [k in keyof Param["properties"]]: FunctionInputRuntimeType<
       Param["properties"][k]
     >;
   }
-  & {
-    [key: string]: UnknownRuntimeType;
-  };
+  : 
+    & {
+      [k in keyof Param["properties"]]: FunctionInputRuntimeType<
+        Param["properties"][k]
+      >;
+    }
```

Comment:
We could probably figure out some way to avoid this duplication, but I didn't want to spend too much time on this. Open to suggestions.